### PR TITLE
Rename `@EnableModule` to `@EnableBinding`

### PR DESCRIPTION
**Pull request by mbogoevici**
Once this is applied, https://github.com/spring-cloud/spring-cloud-stream-modules/pull/30 must be applied as well.

### CODE REVIEW

File: spring-cloud-stream/src/main/java/org/springframework/cloud/stream/annotation/EnableBinding.java [hunk 1]
```diff
@@ -42,10 +43,15 @@
 @Documented
 @Inherited
 @Configuration
-@Import({ChannelBindingServiceConfiguration.class, AggregateBuilderConfiguration.class, BindingBeansRegistrar.class})
+@Import({ ChannelBindingServiceConfiguration.class, AggregateBuilderConfiguration.class,
+		BindingBeansRegistrar.class })
```
Comment (ilayaperumalg) [inline spring-cloud-stream/src/main/java/org/springframework/cloud/stream/annotation/EnableBinding.java]: I believe this line can still fit in the above within the 120 limit.

Comment (mbogoevici) [inline spring-cloud-stream/src/main/java/org/springframework/cloud/stream/annotation/EnableBinding.java]: Agreed. At this juncture, my destiny is controlled by the code formatter. ;)

File: spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BindingBeansRegistrar.java [hunk 2]
```diff
@@ -37,11 +37,11 @@
...
 			BeanDefinitionRegistry registry) {
-		MultiValueMap<String, Object> attributes = metadata.getAllAnnotationAttributes(
-				EnableModule.class.getName(), false);
-		List<String> registeredChannelNames = new ArrayList<>();
+		MultiValueMap<String, Object> attributes = metadata.getAllAnnotationAttributes(EnableBinding.class.getName(),
+				false);
 		for (Class<?> type : collectClasses(attributes.get("value"))) {
-			BindingBeanDefinitionRegistryUtils.registerChannelBeanDefinitions(type, type.getName(), registry);
+			BindingBeanDefinitionRegistryUtils.registerChannelBeanDefinitions(type,
+					type.getName(), registry);
```
Comment (ilayaperumalg) [inline spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BindingBeansRegistrar.java]: Something is weird with your IDE formatter ;) I notice the code format changes in few places...

**Comment by ilayaperumalg** [comment]
At this context does it make sense to rename the `ModuleChannels` annotation to `BindingChannels`?

File: spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BindingBeansRegistrar.java (hunk 2 above)
Comment (mbogoevici) [inline spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BindingBeansRegistrar.java]: Possibly.

**Comment by mbogoevici** [comment]
Possibly. Why not just `@Binding` or `@Bindings` though?

**Comment by ilayaperumalg** [comment]
+1 for `@Bindings`

**Comment by markfisher** [comment]
if that same annotation would eventually apply to other bindable components (e.g. Observable), then +1 for `@Bindings`

**Comment by mbogoevici** [comment]
I think it should ... `@Bindings` is better than `@ModuleChannels` anyway.

**Comment by markfisher** [comment]
rebased, squashed and merged as https://github.com/spring-cloud/spring-cloud-stream/commit/86e457fac41061e75e876779b97e7380c7e5bd88

**Pull request closed without merging**